### ISSUE TITLE
useCombobox: Can't perform a React state update on an unmounted component.

### ISSUE THREAD

**Issue by karlhorky** [downshift-js/downshift#1259]
- `downshift` version: 6.1.1
- `node` version: 15.10.0
- `npm` (or `yarn`) version: yarn 1.22.10

**Relevant code or config**

```javascript
useCombobox({
  onSelectedItemChange: (changes) => {
    if (changes.selectedItem) {
      navigate(changes.selectedItem.path);
    }
  },

  // ...
```

**What you did**:

Navigated to a new page in a Gatsby app in the `useCombobox` `onSelectedItemChange` callback using [`@reach/router`'s `navigate` function](https://reach.tech/router/api/navigate):

**What happened**:

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
```

**Reproduction repository**:

**Problem description**:

Warning about cleanup not being finished. Maybe some effect / state change is still unresolved in the combobox when the `onSelectedItemChange` runs and changes the page, causing unmount of the component?

**Suggested solution**:

Maybe something similar to the solution from @danielstreit in https://github.com/downshift-js/downshift/pull/1256

Although it's not exactly the same, since [`useCombobox` does not have a `debounce`](https://github.com/downshift-js/downshift/issues/1226#issuecomment-809720482)

**Workaround**:

One way of working around it for now is to wrap the `navigate` call in a `setTimeout`, seems to avoid the issue:

```diff
useCombobox({
  onSelectedItemChange: (changes) => {
    if (changes.selectedItem) {
-     navigate(changes.selectedItem.path);
+     setTimeout(() => navigate(changes.selectedItem.path));
    }
  },

  // ...
```

**Comment by silviuaavram** [comment]
There have been some cleanup fixes in 7.0.5, maybe this issue should be fixed as well. Reopen it if not, thanks!

**Comment by karlhorky** [comment]
Seems like some change between `downshift@6.1.1` and `downshift@7.0.5` has indeed resolved the problem - I can no longer reproduce the error.

Thanks for reporting back!
